tests/acceptance/a11y-audit-test.ts

```diff
@@ -137,4 +138,27 @@ module('Acceptance | a11y audit', function (hooks) {
 
     assert.ok(true, 'the image-alt rule should be ignored');
   });
+
+  test('Setting the `enableA11yAudit` query parameter does not mutate the URL', async function (assert) {
+    assert.expect(2);
+
+    const baseUrl = 'https://www.example.com';
+    const url = `${baseUrl}/some/path?foo=123&bar&baz`;
+
+    let href = _calculateUpdatedHref(url, baseUrl, true);
+    let expectedHref = `${url}&enableA11yAudit`;
+    assert.equal(
+      href,
+      expectedHref,
+      'Updated URL matches the expected URL when `enableA11yAudit` is added'
+    );
+
+    href = _calculateUpdatedHref(expectedHref, baseUrl, false);
+    expectedHref = url;
+    assert.equal(
+      href,
+      expectedHref,
+      'Updated URL matches the expected URL when `enableA11yAudit` is removed'
+    );
+  });
```

Comment:
We need a few more test cases:

* when `&enableA11yAudit` is already present and you call `_calculateUpdateHref(url, baseUrl, true)` it shouldn't change anything
* when `&enableA11yAudit` is not present and you call `_calculateUpdateHref(url, baseUrl, false)` it shouldn't do anything
* when `&enableA11yAudit=` is present and you call `_calculateUpdateHref(url, baseUrl, true)` it should normalize to remove the trailing `=` 
* when `&enableA11yAudit=` is present and you call `_calculateUpdateHref(url, baseUrl, false)` it should remove the flag appropriately
* does not normalize other unrelated search params (we cannot assume that **everyone** using any possible query string param wants their `=` removed)

Reply:
I think a small bit of meta-programming will make these test permutations much easier, I'm thinking something like:

```js
  module('enableA11yAudit Query Param', function() {
    const baseUrl = 'https://www.example.com';

    function verify({ label, input, enabled, expected }) {
      test(`${label}: setEnableA11yAudit('${input}', '${baseUrl}', ${enabled}) -> ${output}`, function(assert) {
        let actual = _calculateUpdatedHref(`${baseUrl}/${input}`, baseUrl, expected);

        assert.equal(actual, expected);
      });
    }
    
    verify({
      label: 'can enable',
      input: '/some/path?foo=123&bar&baz',
      enabled: true,
      expected: `/some/path?foo=123&bar&baz&enableA11yAudit`,
    });
    
    verify({
      label: 'can disable',
      input: '/some/path?foo=123&bar&baz&enableA11yAudit',
      enabled: false,
      expected: `/some/path?foo=123&bar&baz`,
    });
    
    verify(
      label: 'already enabled, enabling noops',
      input: '/some/path?foo=123&bar&baz&enableA11yAudit',
      enabled: true,
      expected: `/some/path?foo=123&bar&baz&enableA11yAudit`,
    });
    
    verify({
      label: 'already enabled, enabling normalizes',
      input: '/some/path?foo=123&bar&baz&enableA11yAudit=',
      enabled: true,
      expected: `/some/path?foo=123&bar&baz&enableA11yAudit`,
    });
    
    verify({
      label: 'does not normalize unknown params',
      input: '/some/path?foo=&enableA11yAudit=',
      enabled: true,
      expected: `/some/path?foo=&enableA11yAudit`,
    });
  })
```

Reply:
This seems fine to have the iterative test cases. I'd personally prefer not to have a function like this, but instead just loop with an object of case options. Also, we're testing `_calculateUpdatedHref`, not `setEnableA11yAudit`, so I'd probably reflect that in the test case name.

Reply:
I've beefed up the tests and had to change the param logic considerably to handle the case of not normalizing unrelated params. Let me know how this looks.
<img width="1144" alt="Screen Shot 2021-06-22 at 8 51 43 AM" src="https://user-images.githubusercontent.com/945868/122961374-31617800-d339-11eb-847c-1255165aad6c.png">